File: entrega1/lenguajes_adicionales/cpp/secante.cpp
#include <stdio.h>
#include <math.h>

#define f(x) (exp(-pow(x, 2) + 1) - 4 * pow(x, 3) + 25) // Definimos la funcion de la que se busca la raiz


int main()
{
    double x0 = 1, x1 = 2, xa, fx0 = f(x0), fx1, den, tol = 0.0005, e = tol + 1;
    int iter = 100, n = 0, error = 0;

    printf("|%4s| %10s| %10s| %10s|\n", "n", "xn", "f(x)", "Error");

    if ( fx0 == 0) {
        printf("\nX0 = %f es una Raíz\n", x0);
    } else { 
            fx1 = f(x1);
            den = fx1 - fx0; 
        printf("|%4d| %10f| %10f| %10f|\n", n, x0, fx0, "");
        n++;
        printf("|%4d| %10f| %10f| %10f|\n", n, x1, fx1, "");
        while (e > tol && fx1 != 0 && den != 0 && n < iter - 1) {
            xa = x1 - fx1 * (x1 - x0) / den;
            e = (xa - x1); 
            if(error == 1) e /= x1;
            e = fabs(e); 
            x0 = x1;
            fx0 = fx1;
            x1 = xa;
            fx1 = f(x1);
            den = fx1 - fx0;
            n++;
            printf("|%4d| %10f| %10f| %10f|\n", n, x1, fx1, e);
        }
    }

    printf("\nLa raiz es de f(x) = exp(-x^2+1) - 4x^3 + 25; es: %f\n", x1);

    return 0;
}

File: entrega1/lenguajes_adicionales/swift/python/secante.py
import math

def f(x) : return math.exp(-math.pow(x, 2) + 1) - 4 * math.pow(x, 3) + 25

if __name__ == "__main__":
    x0, x1, tol = 1.0, 2.0,0.0005
    fx0, e = f(x0), tol + 1.0
    fx1, xa, den = None, None, None

    iter, n, error = 100, 0, 0

    print("\n{:^5}{:^18}{:^15}{:^15}\n".format("n", "xn", "f(x)", "Error"))
    if fx0 == 0.0 :
        print("\nX0 = {:.15f} es una Raíz\n".format(x0))
    else :
        fx1 = f(x1);
        den = fx1 - fx0;
        print("|{:^4}|{:^15E}|{:^15E}|{:^15}|".format(n, x0, fx0, ""))
        n += 1
        print("|{:^4}|{:^15E}|{:^15E}|{:^15}|".format(n, x1, fx1, ""))
        while e > tol and fx1 != 0.0 and den != 0.0 and n < iter - 1 :
            xa = x1 - fx1 * (x1 - x0) / den
            e = xa - x1
            if error == 1 :
                e /= x1
            e = abs(e) 
            x0 = x1
            fx0 = fx1
            x1 = xa
            fx1 = f(x1)
            den = fx1 - fx0
            n += 1
            print("|{:^4}|{:^15E}|{:^15E}|{:^15E}|".format(n, x1, fx1, e));

    print("\nLa raiz es de f(x) = exp(-x^2+1) - 4x^3 + 25; es: {:.20f}\n".format(x1))
